docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+        |               |
+    roomStore <---------|
+        ^               |
+        |               |
+    threadStore <---unreadStore
+        ^               |
+        |               |
+    messageStore <------|
+```
+
+In Flux we would need to write these dependencies spread out across 6 different files. In TuxedoJS, we can do the following:
+
+```
+    var userStore = require('../stores/userStore');
+    var roomStore = require('../stores/roomStore');
+    var threadStore = require('../stores/threadStore');
+    var messageStore = require('../stores/messageStore');
+    var privateMessageStore = require('../stores/privateMessageStore');
+    var unreadStore = require('../stores/unreadStore');
+
+    var architect = require('tux/Architecture').architect;
+
+    architect(userStore).itOutputs('users');
+    architect(roomStore).itNeeds('users').itOutputs('rooms', 'unreadCount');
+    architect(threadStore).itNeeds('rooms').itOutputs('threads', 'unreadCount');
+    architect(messageStore).itNeeds('threads').itOutputs('messages', 'unreadCount');
+    architect(privateMessageStore).itNeeds('users').itOutputs('privateMessages', 'unreadCount');
+    architect(unreadStore).itNeeds('unreadCount');
+```
+
+TuxArchitecture also supports arrays of inputs for `itNeeds` and `itOutputs` as well as the ability to chain invocations of these functions via the `and` method (it copies the last method invoked). For example, we could rewrite `architect(roomStore)` as:
+
+```
+    architect(roomStore).itNeeds('users').itOutputs(['rooms', 'unreadCount']); //OR
+    architect(roomStore).itNeeds('users').itOutputs('rooms').and('unreadCount');
+```
+
+Or if we decide that the `unreadStore` was also dependent on the `userStore` we could write:
+
+```
+    architect(unreadStore).itNeeds('unreadCount', 'users'); //OR
+    architect(unreadStore).itNeeds(['unreadCount', 'users']); //OR
+    architect(unreadStore).itNeeds('unreadCount').and('users');
+```
+
+Lastly, you have the option to directly need stores if you do not want to use this string based syntax:
+
+```
+    architect(roomStore).itNeeds(userStore);
+    architect(threadStore).itNeeds(userStore, roomStore);
+    architect(messageStore).itNeeds(userStore, threadStore);
+    architect(privateMessageStore).itNeeds(userStore);
+    architect(unreadStore).itNeeds(roomStore, threadStore, messageStore, privateMessageStore);
+```
+
+While this approach is shorter it can be advantageous to use the string approach as it allows you to write your dependencies in terms of the actual outputs of your stores and thus provides self-documenting code that can make it easier to reason about your application architecture down the line.
+
+Below is the full API documentation for the `architect` method.
+
+####<a id="Architecture-architect"></a>Architecture.architect [#](#Architecture-architect)
+Accepts a store and returns an `architectureChain` instance to define the inputs and outputs for the store.
```

Comment:
Recommend changing sentence to something like "... to define the stores inputs and outputs."